frontend/src/components/Graph.jsx

```diff
@@ -7,19 +7,161 @@ import { useRef, useState, useEffect } from "react";
 import { Link } from "react-router-dom";
 
 const Set = (props) => {
-    const canvasRef = useRef(null);
+  const canvasRef = useRef(null);
 
+  const [workouts, setWorkouts] = useState([]);
+  useEffect(() => {
+    if (props.user) {
+      fetch(`http://localhost:3000/profiles/${props.user.uid}/workouts`)
+        .then((response) => {
+          if (!response.ok) {
+            throw new Error(`HTTP error! status: ${response.status}`);
+          }
+          return response.json(); // Parse JSON data from the response
+        })
+        .then((data) => {
+          // Handle successful response
+          setWorkouts(data);
+          console.log(data);
+        })
+        .catch((error) => {
+          console.error("Error fetching boards:", error);
+        });
+    }
+  }, [props.user]);
+
+  useEffect(() => {
+    const graphRatio = 0.65;
+
+    const canvas = canvasRef.current;
+    const context = canvas.getContext("2d");
+    context.fillStyle = "white";
```

Comment:
Consider extracting this drawing logic into a separate function or custom hook.

Reply:
Sounds good, I will work on cleaning up the graph code for sure!